docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -41,7 +41,74 @@ It uses the following five step approach to accomplish this:
 
 ## User-defined variables
 
-Define your own macro variables with the `@DEF` macro operator. For example, you could set the macro variable `macro_var` to the value `1` with:
+SQLMesh supports two kinds of user-defined macro variables: global and local.
+
+Global macro variables are defined in the project configuration file and can be accessed in any project model.
+
+Local macro variables are defined in a model definition and can only be accessed in that model.
+
+### Global variables
+
+Global variables are defined in the project configuration file [`variables` key](../../reference/configuration.md#variables).
+
+Global variable values may be any of the following data types or lists or dictionaries containing these types: `int`, `float`, `bool`, `str`.
+
+Access global variable values in a model definition using the `@<VAR_NAME>` macro or the `@VAR()` macro function. The latter function requires the name of the variable _in single quotes_ as the first argument and an optional default value as the second argument. The default value is a safety mechanism used if the variable name is not found in the project configuration file.
+
+For example, this SQLMesh configuration key defines six variables of different data types:
+
+```yaml linenums="1"
+variables:
+    int_var: 1
+    float_var: 2.0
+    bool_var: true
+    str_var: "cat"
+    list_var: [1, 2, 3]
+    dict_var:
+        key1: 1
+        key2: 2
+```
+
+Additionally, variables can be configured as part of the gateway, in which case individual variable values take precedence over values with the same key in the root configuration:
+```yaml linenums="1"
+gateways:
+    my_gateway:
+        variables:
+            my_var: 1
+        ...
+```
+
+A model definition could access the `int_var` value in a `WHERE` clause like this:
+
+```sql linenums="1"
+SELECT *
+FROM table
+WHERE int_variable = @INT_VAR
```

Comment:
do we note anywhere in the docs that vars are always case insensitive? if not i think we should